ide/git/src/org/netbeans/modules/git/ui/push/PushWizard.java

```diff
@@ -169,7 +169,7 @@ public synchronized void nextPanel () {
                 Collection<PushMapping> mappings = pushBranchesStep.getSelectedMappings();
                 Map<String, String> remoteBranches = new LinkedHashMap<String, String>(mappings.size());
                 for (PushMapping mapping : mappings) {
-                    if (!mapping.isDeletion() && mapping instanceof PushMapping.PushBranchMapping) {
+                    if ((!mapping.isDestructive() || mapping.getLocalName() != null) && mapping instanceof PushMapping.PushBranchMapping) {
```

Comment:
is this a bugfix?

Reply:
I thought that I could reconstruct the reason, but all test cases I am able to construct now don't match the case that is added by this. I'm pretty sure, that I hit that code path, but fail to recall.

Reply:
Ok, I finally managed to reproduce it. After a push the `<remote>/<branch>` branches are updated to the repository state. After a force push this update is also destructive for that mirror branch. This case could not happen before, because the push would have been rejected and thus there would be nothing to pull.